### PR TITLE
welcome-feature: add routing logic

### PR DESCRIPTION
**Pull request by 5h1rU**
### Description

Adds routing logic to the new views in the post-sign-up flow.
Whenever the new user signs up - they'll navigate to the new `welcome` route.

### CODE REVIEW

File: client/web/src/routes.tsx [hunk 1]
```diff
@@ -123,8 +123,31 @@ export const routes: readonly LayoutRouteProps<any>[] = [
...
+             * 2. it's a DotComMode instance
+             * AND
+             * instance has enabled enablePostSignupFlow experimental feature
+             * OR
+             * user authenticated has a AllowUserViewPostSignup tag
+             */
+
+            !!props.authenticatedUser &&
+            window.context.sourcegraphDotComMode &&
+            ((props.authenticatedUser && window.context.experimentalFeatures.enablePostSignupFlow) ||
```
Comment (tjkandala) [inline client/web/src/routes.tsx:140]: Why do we check `props.authenticatedUser` again? Shouldn't it always be defined at this point?

**Review by tjkandala — APPROVED** [review]

File: client/web/src/routes.tsx (hunk 1 above)
Comment (limitedmage) [inline client/web/src/routes.tsx:140]: Can we use the new a/b testing capable feature flags instead of `experimentalFeatures` here?

Comment (artemruts) [reply client/web/src/routes.tsx]: Good catch!

**Review by ryanslade — APPROVED** [review]

**Review by unknwon — APPROVED** [review]
Go code LGTM!

File: client/web/src/routes.tsx (hunk 1 above)
Comment (artemruts) [reply client/web/src/routes.tsx]: I'll try swapping for a/b features after we do the final round of QA. Thank you!

**Pull request merged**